File: src/me/xerces/eventbus/IEventManager.java
package me.xerces.eventbus;

import me.xerces.eventbus.events.Event;

/**
 * @author Xerces
 * @since 18/01/2015
 */
public interface IEventManager {

    public void addEventListener(Object object);

    public void addSpecificEventListener(Object object, Class<?extends Event> eventClass);

    public void removeEventListener(Object object);

    public void fireEvent(Event event);

}

File: src/main/java/dev/throwouterror/eventbus/IEventManager.kt
package dev.throwouterror.eventbus

import dev.throwouterror.eventbus.events.Event

interface IEventManager {
    fun addEventListener(`object`: Any)
    fun addSpecificEventListener(`object`: Any, eventClass: Class<out Event?>)
    fun removeEventListener(`object`: Any?)
    fun fireEvent(event: Event)
}